pkg/collect/cluster_resources.go

```diff
@@ -443,23 +440,45 @@ func crs(
 		}
 		apiResourceList, _ := apiResourceListObj.(*metav1.APIResourceList)
 		groupVersion := apiResourceList.GroupVersion
-		for _, v := range apiResourceList.APIResources {
-			customResourceName := v.Name
-			if customResourceName != "" && !strings.ContainsAny(customResourceName, "/") {
-				fileName := customResourceName + "." + group + ".json"
-				customResourcesResponse, err := client.RESTClient().Get().AbsPath("/apis/" + groupVersion).Namespace("").Resource(customResourceName).DoRaw(ctx)
-				if err != nil {
-					errorList[fileName] = err.Error()
-					continue
-				}
-				_ = json.Unmarshal(customResourcesResponse, &customResourceItems)
-				if len(customResourceItems.Items) != 0 {
-					b, err := json.MarshalIndent(customResourceItems.Items, "", "  ")
-					if err != nil {
-						errorList[fileName] = err.Error()
-						continue
+
+		for _, namespace := range namespaces {
+			for _, resource := range apiResourceList.APIResources {
+				customResourceName := resource.Name
+				if resource.Namespaced {
+					// Avoid subresources
+					if !strings.ContainsAny(customResourceName, "/") {
+						customResourcesResponse, err := client.RESTClient().Get().AbsPath("/apis/" + groupVersion).Namespace(namespace).Resource(customResourceName).DoRaw(ctx)
+						if err != nil {
+							errorList[fmt.Sprintf("%s/%s", group, namespace)] = err.Error()
+							continue
+						}
+						_ = json.Unmarshal(customResourcesResponse, &customResourceItems)
+						if len(customResourceItems.Items) != 0 {
+							b, err := json.MarshalIndent(customResourceItems.Items, "", "  ")
+							if err != nil {
+								errorList[fmt.Sprintf("%s/%s", group, namespace)] = err.Error()
+								continue
+							}
+							customResources[fmt.Sprintf("%s/%s.json", group, namespace)] = b
```

Comment:
This is reverting the agreed file tree structure and removing resource kind from the directory name.

Reply:
Fixed. Now the tree structure looks like
```
└── elasticsearches.elasticsearch.k8s.elastic.co
    ├── default.json
    ├── prod.json
    └── test.json
```
instead of 
```
└── elasticsearch.k8s.elastic.co
    ├── default.json
    ├── prod.json
    └── test.json
```